### DISCUSSION TITLE
xray-junit-extensions (Linking Tests to a specific Test Key)

### DISCUSSION THREAD

**Discussion by shrubom** [Xray-App/xray-junit-extensions#46] [Show and tell]
Hi,
I would like to know if we can manually create Test Issue Type (generic) {Say, TEST-1234 -> Type Generic}.
Now I want to link a Test to the above mentioned Test Key.

Code Snippet:

    @Test  
    @XrayTest(key = "TEST-1234")
    @Tags({@Tag("Regression"), @Tag(""TEST-1234")})
    public void validateErrMessageForForeignCountry()  {
     xrayReporter.addComment("This test Validates error message");
     // Test code        
    }
    
Upon running the build on Jenkins:
Summary, Description, Comments and Tags are working but the Link to the Test Key does not work. It still creates a new Test Key.

Could you please advice?

**Comment by bitcoder** [comment]
Can you please the generate JUnit XML report? (not the surefire but the one generated by the listener of this project).
Note: you just need to add the @XrayTest annotation; no need to add the test issue key as a tag.
Can you please share the output of Jenkins build and a screnshot of your existing Test issue in Xray (e.g., TEST-1234)?

**Reply by shrubom** [reply]
I removed the Test Issue Key number from the @Tag annotation and now everything is working as expected. The results are reflecting to the correct Test Key. And no new test case created. :)

So now my code looks like this:
```
@Test  
@XrayTest(key = "TEST-1234")
@Tags({@Tag("Regression")})
public void validateErrMessageForForeignCountry()  {
 xrayReporter.addComment("This test Validates error message");
 // Test code        
}
```

Thank you so much for the prompt response.
